### PR TITLE
feat(input-date-picker, input-time-picker): add status property

### PR DESCRIPTION
**Pull request by benelan**
**Related Issues:** #5723, #4276

## Summary

Add `status` property to `calcite-input-date-picker` and `calcite-input-time-picker`. Validation messages will be added in a second pass to ensure consistency with the messages displayed for components with invalid values on form submission.

### CODE REVIEW

File: packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts [hunk 1]
```diff
@@ -20,110 +20,44 @@ export const simple = (): string => html`
...
-    ${boolean("hidden", false)}
-    name="${text("name", "light")}"
-    scale="${select("scale", ["s", "m", "l"], "m")}"
-    step="${number("step", 1)}"
-    value="${text("value", "10:37")}"
-    lang="ko"
-    open
-  >
-  </calcite-input-time-picker>
+export const koreanLocaleInvalidStatus_TestOnly = (): string => html`
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts:55]: Should the invalid status be a separate story to keep this one strictly focused on `ko`?

File: packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts [hunk 2]
```diff
@@ -20,110 +20,44 @@ export const simple = (): string => html`
...
-    ${boolean("disabled", false)}
-    ${boolean("hidden", false)}
-    class="calcite-mode-dark"
-    name="${text("name", "dark")}"
-    scale="${select("scale", ["s", "m", "l"], "m")}"
-    step="${number("step", 1)}"
-    value="${text("value", "22:37")}"
-  >
-  </calcite-input-time-picker>
+  <calcite-input-time-picker class="calcite-mode-dark" value="22:37" step="1"> </calcite-input-time-picker>
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts:46]: ✨🧹✨

File: packages/calcite-components/src/components/input-number/input-number.stories.ts [hunk 3]
```diff
@@ -16,7 +16,6 @@ export const simple = (): string => html`
   <div style="width:300px;max-width:100%;text-align:center;">
     <calcite-input-number
       scale="${select("scale", ["s", "m", "l"], "m")}"
-      status="${select("status", ["idle", "valid", "invalid"], "idle")}"
```
Comment (jcfranco) [inline packages/calcite-components/src/components/input-number/input-number.stories.ts:19]: Can you add a story for the `invalid` status?

**Review by driskull — APPROVED** [review]
👍

**Review by jcfranco — APPROVED** [review]
Other than some test tweak suggestions, this LGTM! 🎉

File: packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts:55]: Let's add a separate one. 

I agree we need to decrease snapshot count. Can you create an issue to do another pass to identify and consolidate stories? I think we could do so for anything testing different locales and scales.

Comment (benelan) [reply packages/calcite-components/src/components/input-time-picker/input-time-picker.stories.ts:55]: Sure. We can also combine multiple components with different states into a single story. Like a "locale" story with 5 different inputs of various locales and a "scale" story with all three scales.

**Pull request merged**